<a href="https://colab.research.google.com/github/pullz6/Two_Sample_Testing_on_gaming_profiles/blob/main/Testing_hypothesis_on_gaming_profiles.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Hypothesis Testing with online Gamers

We are going to test an assumption that I have of online gamers in general. Generally, I believe that gamers who identify as female tend to gravitate towards simulation games. We will test the null hypothesis against the alternative hypothesis.

Null Hypothesis = There is no relationship between gender and game preference.

Alternative Hypothesis = There is a relationship between gender and game.

In [10]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import scipy.stats as stats
import math

from google.colab import drive
drive.mount('/content/drive')
df = pd.read_csv("/content/drive/My Drive/sample_data/online_gaming_behavior_dataset.csv")

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


In [11]:
df

PlayerID  Age  Gender Location   GameGenre  PlayTimeHours  \
0          9000   43    Male    Other    Strategy      16.271119   
1          9001   29  Female      USA    Strategy       5.525961   
2          9002   22  Female      USA      Sports       8.223755   
3          9003   35    Male      USA      Action       5.265351   
4          9004   33    Male   Europe      Action      15.531945   
...         ...  ...     ...      ...         ...            ...   
40029     49029   32    Male      USA    Strategy      20.619662   
40030     49030   44  Female    Other  Simulation      13.539280   
40031     49031   15  Female      USA         RPG       0.240057   
40032     49032   34    Male      USA      Sports      14.017818   
40033     49033   19    Male      USA      Sports      10.083804   

       InGamePurchases GameDifficulty  SessionsPerWeek  \
0                    0         Medium                6   
1                    0         Medium                5   
2                    0           Easy               16   
3                    1           Easy                9   
4                    0         Medium                2   
...                ...            ...              ...   
40029                0           Easy                4   
40030                0           Hard               19   
40031                1           Easy               10   
40032                1         Medium                3   
40033                0           Easy               13   

       AvgSessionDurationMinutes  PlayerLevel  AchievementsUnlocked  \
0                            108           79                    25   
1                            144           11                    10   
2                            142           35                    41   
3                             85           57                    47   
4                            131           95                    37   
...                          ...          ...                   ...   
40029                         75           85                    14   
40030                        114           71                    27   
40031                        176           29                     1   
40032                        128           70                    10   
40033                         84           72                    39   

      EngagementLevel  
0              Medium  
1              Medium  
2                High  
3              Medium  
4              Medium  
...               ...  
40029          Medium  
40030            High  
40031            High  
40032          Medium  
40033          Medium  

[40034 rows x 13 columns]

In [12]:
df['Location'].unique()

array(['Other', 'USA', 'Europe', 'Asia'], dtype=object)

In [24]:
#Let's get the women's dataset
df_f = df[df['Gender']=='Female']
#Getting the sample set with simulation for women
df_fs = df_f[df_f['GameGenre'] == 'Simulation']

#Let's get the men's dataset
df_m = df[df['Gender']=='Male']
#Getting the sample set with simulation for men
df_ms = df_m[df_m['GameGenre'] == 'Simulation']

In [33]:
female_playing_times = df_fs['PlayTimeHours']
male_playing_times = df_ms['PlayTimeHours']

female_sw = df_fs['SessionsPerWeek']
male_sw = df_ms['SessionsPerWeek']

# Two Sample Testing
Here, we are going to test the null hypothesis on both groups. We will first get the means of female and male gamers, then we will use the stats.ttest_ind() function.

In [34]:
stats.ttest_ind(a= female_playing_times,
                b= male_playing_times,
                equal_var=False)

TtestResult(statistic=1.5618365407651824, pvalue=0.11837233573556608, df=6903.862664091727)

So the p value is 0.11, there is a general threshold of 0.05, this means that we have to accept the null hypothesis, indicating that there is no difference between the playing times in simulation games for male and female.

Lets also check with regards to the sessions per week.

In [35]:
stats.ttest_ind(a= female_sw,
                b= male_sw,
                equal_var=False)

TtestResult(statistic=0.29209510884598683, pvalue=0.7702228064495109, df=6808.133922948998)

As you can see the p valye is 0.77 which is more than 0.05 meaning that again we have to accept the null hypothesis again.